### ISSUE TITLE
Custom modules appear in Studio but nowhere else

### ISSUE THREAD

**Issue by philippmeixner** [salesagility/SuiteCRM-Core#55]
#### Issue
I created and deployed 8 custom modules in our new 8.0.0 installation using the Module Builder, and they all show up in Studio. The database has tables for every module. But 5 of those modules are not displayed in the menu bar, and there is no entry for them under “Display Modules and Subpanels” or " Configure Module Menu Filters". They are also not in “Rename Modules”.

The other 3 custom modules (from the same package and deployment!) are working fine. If I manually modify the URL, I can access the listview of those ‘lost’ modules, but the design looks different and the buttons are missing.

The modules are all of different types (Basic, Company, etc.) so I don’t think that matters.

The logs show the following: 
```
[3185550][1][WARN] ModuleNameMapper | mapName | 'SMP01_servicecontracts' not mapped to 'frontend'
[3185550][1][WARN] ModuleNameMapper | toLegacy | 'SMP01_servicecontracts' not mapped
```
There are entries like this for every custom module I deployed, even from other packages. 

#### Expected Behavior
All deployed custom modules should be available in the following menus: “Display Modules and Subpanels”, " Configure Module Menu Filters", "Rename Modules"

#### Possible Fix
Fix mapping in ModuleNameMapper?

#### Steps to Reproduce
Those steps are not proven to reproduce the issue, but it is what we did before the issue appeared:  
1. Create package in ModuleBuilder and fill with custom modules
2. Deploy package in same instance
3. Add custom fields to those modules in Studio
4. Set up relationships
5. Notice that some modules are missing in other menus and top menu bar

#### Your Environment
* SuiteCRM Version used: 8.0.1
* Browser name and version: Chrome 97.0.4692.71 (Official Build) (64-bit)
* Environment name and version: MySQL, PHP-FPM 7.4, nginx
* Operating System and version: Ubuntu 20.04.2 LTS

**Comment by scicali** [comment]
SuiteCRM uses cache extensively so:

use "Quick Repair and Rebuild" in admin->Repair Section
exec ./bin/console cache:clear from cli, also
use browser in dev mode, with cache disabled
refresh the page

remember to update to 8.0.1

**Comment by philippmeixner** [comment]
Thank you for the rapid response. We just did the following:

- Cleared cache of environment using ```./bin/console cache:clear```
- Upgraded to 8.0.1 with no errors
- Quick Repair and Rebuild

But the problem persists. My browser and the crm system are in dev mode, with cache disabled.

**Comment by scicali** [comment]
I'm not a SuiteCRM developer, I'm a user. But I think is not a bug, I deployed lot of modules, lot of times in the last months, without issues.

Can be a folder permission issue? After installation you must re-set folder permission

**Comment by philippmeixner** [comment]
After a week of troubleshooting I was able to identify the issue. When I created the modules in Module Builder, I disabled the option **Navigation Tab**, as I was thinking this is the same as hiding the modules in _Display Modules and Subpanels_. I was wrong, because this option completely disabled the module from ever appearing in the user interface. I was able to correct this by editing the module definitions in `custom/Extension/application/Ext/Include/PackageName.php`.

[I added to the documentation](https://github.com/salesagility/SuiteDocs/pull/565) so others don't make the same mistake.
